### ISSUE TITLE
Type system in exported JSON should be minimized

### ISSUE THREAD

**Issue by reckart** [inception-project/inception#3422]
By default, only a minimal type system should be included (i.e. one that covers only the types actually used in the respective JSON file). However, I didn't have the time to investigate yet why the included type system is not minimized and if that is a bug/missing setting in INCEpTION or in the underlying UIMA library.

_Originally posted by @reckart in https://github.com/inception-project/inception/issues/3402#issuecomment-1266583149_
      
-----

* Upstream issue: https://github.com/apache/uima-uimaj-io-jsoncas/issues/13

**Comment by jfiala** [comment]
Maybe you could use a json schema and export that as well in a zip file.
That worked great for me, for XMI (as the TypeSystem.xml is always the same).

**Comment by jfiala** [comment]
The type system of exported XMI could be also minimized IMHO - currently it contains TagsetDescription/FeatureDefinition which doesn't seem to be 100% necessary (as it is also contained in the TypeSystem.xml).

Example TagsetDescription:
XMI:
```
<type3:TagsetDescription xmi:id="307" sofa="1" begin="0" end="0" layer="webanno.custom.BISKOMP" name="Match level" input="false"/>
```

TypeSystem.xml:
```
        <typeDescription>
                        
            <name>webanno.custom.BISKOMP</name>
                        
            <description/>
                        
            <supertypeName>uima.tcas.Annotation</supertypeName>
                        
            <features>
                                
                <featureDescription>
                                        
                    <name>identifier</name>
                                        
                    <description/>
                                        
                    <rangeTypeName>uima.cas.String</rangeTypeName>
                                    
                </featureDescription>
                                
...
```

Example FeatureDefinition:
```
<type:FeatureDefinition xmi:id="4804" layer="4793" name="quality" uiName="quality"/>
``` 

However, feature quality is also described attached to its layer in the TypeSystem.xml:
```
 <featureDescription>
                                        
                    <name>quality</name>
                                        
                    <description/>
                                        
                    <rangeTypeName>uima.cas.Float</rangeTypeName>
                                    
                </featureDescription>
                            
            </features>
```

**Comment by reckart** [comment]
The `FeatureDefinition` that is included in the XMI is supplemental to the definitions in the UIMA type system. The UIMA type system has no concept of a `display name` for a feature. It has a `description` and a `name`, but no `display name`. The `name` of a feature cannot be changed. The `display name` of a feature can be changed in INCEpTION. Some people had the problem that they had changed the `display name` in INCEpTION to something wildly different from the `name` they set during the creation of the feature. Thus, `FeatureDefinition` was added to provide them with access to the `display name`.

**Comment by jfiala** [comment]
Thanks for the explanation, got it :) (uiName = display name).

**Comment by reckart** [comment]
Fixed in e730f72ac5021b87d3f778e85781066b2a5edf74 as part of #3571 .
